src/Core/Configure/Engine/PhpConfig.php

```diff
@@ -92,9 +92,37 @@ public function read($key)
      */
     public function dump($key, array $data)
     {
-        $contents = '<?php' . "\n" . 'return ' . var_export($data, true) . ';';
+        $contents = '<?php' . "\n" . 'return ' . $this->shortArrayVarExport($data) . ';';
 
         $filename = $this->_getFilePath($key);
         return file_put_contents($filename, $contents) > 0;
     }
+
+    /**
+     *
+     * Converts an array $data into a string of PHP code using the Short Array Syntax.
+     *
+     * Similar to var_export() except it uses the Short Array Syntax;
+     *
+     * @param array $data Data to dump in Short Array Syntax
+     * @return string A parsable string representation of a short array syntax variable
+     */
+    public function shortArrayVarExport($data)
+    {
+        $varExport = var_export($data, true);
+        $pattern = [
+            '/array \(/', // Matches opening "array("
+            '/(\),)$/m', // Matches ending "),"
+            '/^\)$/m' // Matches last ) in file
```

Comment:
These patterns could also occur in string data. Replacing these patterns there could corrupt configuration data.